types/jasmine/index.d.ts

```diff
@@ -247,11 +248,11 @@ declare namespace jasmine {
     function objectContaining<T>(sample: Partial<T>): ObjectContaining<T>;
     function createSpy(name?: string, originalFn?: Function): Spy;
 
-    function createSpyObj(baseName: string, methodNames: SpyObjMethodNames): any;
-    function createSpyObj<T>(baseName: string, methodNames: SpyObjMethodNames<T>): SpyObj<T>;
+    function createSpyObj(baseName: string, methodNames: SpyObjMethodNames, propertyNames?: SpyObjMethodNames): any;
+    function createSpyObj<T>(baseName: string, methodNames: SpyObjMethodNames<T>, propertyNames?: SpyObjMethodNames<T>): SpyObj<T>;
```

Comment:
It's incorrect to use `SpyObjMethodNames<T>` here, as it will kill types for the properties. Instead, you should create a new type for this purpose:

```ts
    type SpyObjPropertyNames<T = unknown> =
        unknown extends T ?
            (ReadonlyArray<string> | { [propertyName: string]: any }) :
            (ReadonlyArray<keyof T> | { [ P in keyof T ]?: T[P] } )
```